go/private/rules/binary.bzl

```diff
@@ -46,6 +46,19 @@ load(
 
 _EMPTY_DEPSET = depset([])
 
+def _include_path(hdr):
+    if not hdr.root.path:
+        fail("Expected hdr to be a generated file, got source file: " + hdr.path)
+
+    root_relative_path = hdr.path[len(hdr.root.path + "/"):]
+    if not root_relative_path.startswith("external/"):
```

Comment:
Will this work with `--experimental_sibling_repository_layout`?

Reply:
Yes, it does: With the layout, the repository name will be contained in the root. The `root_relative_path` will not start with `external`, so all that will be returned is `hdr.root.path`.

I added a test case for this.